src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoN.java

```diff
@@ -0,0 +1,483 @@
+     *
+     * @param file      the underlying HDF5 file.
+     * @param logger    the logger to log the warning message with.
+     * @throws IllegalArgumentException if {@code file} is {@code null}.
+     */
+    public HDF5PCACoveragePoN(final HDF5File file, final Logger logger) {
+        this(file);
+        if (getVersion() < CURRENT_PON_VERSION) {
+            logger.warn("The version of the specified PoN (" + getVersion() + ") is older than the latest version " +
+                    "(" + CURRENT_PON_VERSION + ").");
+        }
+    }
+
+    @Override
+    public double getVersion() {
+        return file.readDouble(VERSION_PATH);
+    }
+
+    @Override
+    public List<Target> getTargets() {
+        return targets.get();
+    }
+
+    @Override
+    public List<Target> getRawTargets() {
+        return rawTargets.get();
+    }
+
+    @Override
+    public List<Target> getPanelTargets() {
+        return panelTargets.get();
+    }
+
+    @Override
+    public double[] getTargetFactors() {
+        final double[] values = file.readDoubleArray(TARGET_FACTORS_PATH);
+        if (values.length != targetNames.get().size()) {
+            throw new GATKException(String.format("Wrong number of elements in the target factors recovered from file '%s': %d != %d", file.getFile(), values.length, targetNames.get().size()));
+        }
+        return values;
+    }
+
+    @Override
+    public double[] getTargetVariances() {
+        final double[] values = file.readDoubleArray(TARGET_VARIANCES_PATH);
+        if (values.length != panelTargets.get().size()) {
+            throw new GATKException(String.format("wrong number of elements in the target variances recovered from file '%s': %d != %d", file.getFile(), values.length, panelTargets.get().size()));
+        }
+        return values;
+    }
+
+    @Override
+    public RealMatrix getNormalizedCounts() {
+        // Note the check uses sample names as number of rows and targets as number of columns.  This is due to the
+        //  transposed storage.  The returned matrix is still targets (rows) x samples (columns).
+        return file.readMatrixAndCheckDimensions(NORMALIZED_COUNTS_PATH, sampleNames.get().size(), targetNames.get().size())
+                .transpose();
+    }
+
+    @Override
+    public RealMatrix getLogNormalizedCounts() {
+        // Note the check uses sample names as number of rows and targets as number of columns.  This is due to the
+        //  transposed storage.  The returned matrix is still targets (rows) x samples (columns).
+        return file.readMatrixAndCheckDimensions(LOG_NORMALIZED_COUNTS_PATH, getPanelSampleNames().size(), getPanelTargetNames().size())
+                .transpose();
+    }
+
+    @Override
+    public RealMatrix getLogNormalizedPInverseCounts() {
+        return file.readMatrixAndCheckDimensions(LOG_NORMALIZED_PINV_PATH, getPanelSampleNames().size(),
+                getPanelTargetNames().size());
```

Comment:
right, this is mentioned in the javadoc for the class but i can add comments here.

also, just realized that some of the dimensions in the PCACoveragePoN javadoc are also wrong, will fix these as well.